Cython/Utility/ModuleSetupCode.c

```diff
@@ -1019,6 +1105,12 @@ static CYTHON_SMALL_CODE PyObject* ${pymodule_create_func_cname}(PyObject *spec,
     Py_DECREF(modname);
     if (unlikely(!module)) goto bad;
 
+#if CYTHON_COMPILING_IN_LIMITED_API
+    if (unlikely(PyModule_AddStringConstant(module, "loader", "__loader__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "origin", "__file__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "parent", "__package__") < 0)) goto bad;
+    if (unlikely(PyModule_AddStringConstant(module, "submodule_search_locations", "__path__") < 0)) goto bad;
```

Comment:
Ah, no, this is not what the original code did. It actually copied the value of an attribute from the `ModuleSpec` over to the module. Why can't the limited-API code do that as well?

Reply:
Woops, misinterpreted what the function was doing. You are right this is incorrect, I'll fix it. I'll add a TODO for now, I want to start integrating into the CI first.

The reason the Limited API can't support this is because the previous code was pulling the module dict out. In the Limited API world you should never pull any instance's internal dictionary out to manipulate it. Instead you need to rely on the Limited APIs to do this. In this case, we can use `PyModule_AddObject`.

Reply:
Fixed, this now updates `__Pyx_copy_spec_to_module ` to do `PyModule_AddObject ` when using the Limited API